### PR TITLE
Satisfy degree reqs

### PR DESCRIPTION
**Pull request by kayleebement**
Adding / removing courses satsifies / unsatisfies requirements for normal degrees!

Also fixed some small bugs with course modal (switching degrees wasn't refreshing course + unit dropdowns) and requirements.json (engineering fundamentals were formatted incorrectly)

### CODE REVIEW

**Comment by kayleebement** [comment]
just merged master into this branch and tested, still workin

**Comment by kayleebement** [comment]
Riley found bug w track requirement B, fixing now

**Comment by kayleebement** [comment]
all fixed :)

**Comment by kayleebement** [comment]
NOTE: I also added in the new degrees into the DB, and since we dont' have track support right now, some degrees (like CS) look really wild..... 

NOTE: sometimes if you don't wait long enough for everything to load, adding a degree and then trying to open it will crash it. issue for adding loading graphic is up, and this problem is present in master anyways so shouldn't stop us from merging

**Comment by kayleebement** [comment]
merged new master into branch, going to test and make sure merge worked

**Comment by kayleebement** [comment]
everything looks like it's still working, also rn you can't satisfy things like WIM and Senior Project with same course,  this is expected and will be possible once sidebar popup isa thing (to override reqs)

**Comment by kayleebement** [comment]
tried to merge the newest master and now everything is broken. ignore for now.

**Comment by kayleebement** [comment]
and we are back and functioning!

File: frontend/app/src/components/course-modal/course-modal.js [hunk 1]
```diff
@@ -75,9 +81,20 @@ class Modal extends Component {
     selected department is saved and the dropdown for courses is displayed;
     otherwise, selected course set to "".
   */
+
   selectDepartment(dep) {
     const department = dep.label
     this.setState({selectedDepartment: department})
+    console.log(this.state.selectedDepartment)
```
Comment (kate-rydberg) [inline frontend/app/src/components/course-modal/course-modal.js]: remove

Comment (kate-rydberg) [inline frontend/app/src/components/course-modal/course-modal.js]: remove

File: frontend/app/src/components/course-modal/course-modal.js [hunk 2]
```diff
@@ -86,6 +103,7 @@ class Modal extends Component {
     var courseData = this.props.allCourseDict[this.state.selectedCourseID];
     const course = {
       id: this.state.count + this.state.selectedCourseID, // TODO: what if you retake a course? Need to be more distinct
+      course_id: this.state.selectedCourseID, // need ID without count
```
Comment (kate-rydberg) [inline frontend/app/src/components/course-modal/course-modal.js]: for retaking, what if they just drag class from original quarter to new quarter?

Comment (kayleebement) [reply frontend/app/src/components/course-modal/course-modal.js]: that wasn't my comment/ID setup in the first place, I agree with what you said
I only added the course_id

File: frontend/app/src/components/course-modal/course-modal.js (hunk 1 above)
Comment (kayleebement) [reply frontend/app/src/components/course-modal/course-modal.js]: done

File: frontend/app/src/components/course-modal/course-modal.js (hunk 2 above)
Comment (kate-rydberg) [reply frontend/app/src/components/course-modal/course-modal.js]: oh i totally missed that - sorry about that!!

**Comment by eliew88** [comment]
I have been reading through the code, and looks good to me! Am not able to run it currently due to git issues, but I at least approve with my visual inspection.

**Pull request merged**